### ISSUE TITLE
Write a pre-commit script

### ISSUE THREAD

**Issue by mristin** [pschanely/CrossHair#57]
Maybe this has been already done, but I can not find it -- do you use a code formatter for the project? I only see that you check flake8 on Travis, but can not find any script that would automatically format the code.

Would you like me to set up a pre-commit script, akin to [this one](https://github.com/mristin/icontract-hypothesis/blob/main/precommit.py) so that the code can be automatically formatted, test automatically run, docstrings checked *etc.*?

**Comment by pschanely** [comment]
As you've deduced, there are a **lot** of things to work on in terms of coding standards. Heck, mypy doesn't even pass on everything yet.

There is no formatter currently; the flake8 settings only detect super bad things like undefined symbols etc. It would be great if you wanted to take a stab at a mass formatting and a hook to enforce/help with that. I've been trying to keep recent edits to an 80 char width, but plenty of code doesn't even have that.

**Comment by mristin** [comment]
Would it be possible to go with 100 characters as line width? I just ran `black` over the code and there are hundreds of >80 char lines. However, there are ~100-200 lines with >100 char lines.

The output of `black` also looked reasonable on 100 line width upon visual inspection, so that's what I would suggest unless there is a really compelling reason for 80 char line width.

(I'll create a pull request with the pre-commit script once the pull request #58 is merged in to avoid conflicts.)

**Comment by pschanely** [comment]
100 chars sounds good to me!

**Comment by mristin** [comment]
Fixed by #71 .
